lib/recommender_system.py

```diff
@@ -12,14 +12,16 @@
 from util.top_recommendations import TopRecommendations
 from util.data_parser import DataParser
 from util.recommender_configuer import RecommenderConfiguration
+from util.model_initializer import ModelInitializer
 
 
 class RecommenderSystem(object):
     """
     A class that will combine the content-based and collaborative-filtering,
     in order to provide the main functionalities of recommendations.
     """
-    def __init__(self, abstracts_preprocessor=None, ratings=None, process_parser=False, verbose=False):
+    def __init__(self, initializer=None, abstracts_preprocessor=None, ratings=None,
```

Comment:
Add dump option, docs for the new variables, and preferably a better name for reinit.